thetis/shallowwater_eq.py

```diff
@@ -317,8 +329,7 @@ def residual(self, uv, eta, uv_old, eta_old, fields, fields_old, bnd_conditions=
                     # assume land boundary
                     # impermeability implies external un=0
                     un_jump = inner(uv, self.normal)
-                    h = self.bathymetry
-                    head_rie = head + sqrt(h/g_grav)*un_jump
+                    head_rie = head + sqrt(total_h/g_grav)*un_jump
```

Comment:
This changes behavior. Previously, even with `nonlin=True` we set `h=bathymetry` which implies that the factor in front of the jump is independent of the simulation state (`eta`). Because `total_h` is defined with `eta_old` the proposed modification might be OK (the jump term is still linear), but we need to make sure that everything works.